### ISSUE TITLE
TLS rule presents the same result multiple times. 

### ISSUE THREAD

**Issue by gcmurphy** [HewlettPackard/gas#8]
Because by default all rules are run, and as there are three variants of the TLS rule whenever we get a match for cipher suites we get the same error message multiple times with little distinction between them. 

For example given this test file: 

```go
package main

import (
	"crypto/tls"
	"fmt"
	"net/http"
)

func main() {
	tr := &http.Transport{
		TLSClientConfig: &tls.Config{MinVersion: 0},
	}
	client := &http.Client{Transport: tr}
	_, err := client.Get("https://golang.org/")
	if err != nil {
		fmt.Println(err)
	}
}
```

Executing gas gives the same error three times: 

```
$ ../gas  ./min_version.go
Results:

[./min_version.go:11] - TLS MinVersion too low. (Confidence: HIGH, Severity: HIGH)
  > MinVersion: 0


[./min_version.go:11] - TLS MinVersion too low. (Confidence: HIGH, Severity: HIGH)
  > MinVersion: 0


[./min_version.go:11] - TLS MinVersion too low. (Confidence: HIGH, Severity: HIGH)
  > MinVersion: 0


Summary:
   Files: 1
   Lines: 18
   Nosec: 0
  Issues: 3
```

I think we need to perhaps revisit how we are registering rules and selecting defaults. Perhaps we need something like profiles?

**Comment by callidus** [comment]
yeah, I think its time to take a look at that. The TLS stuff should really be three levels that you can select between rather than three checks. I guess we need some more advanced config stuff (profiles as you suggest).

**Comment by gcmurphy** [comment]
@callidus Is this resolved now via rule ID's? We probably can fix any remaining use #15?

**Comment by gcmurphy** [comment]
I'm going to close this out for now. The remaining configuration of TLS profiles can ben fixed by other open TLS issue.
